cmd/create/idp/cmd.go

```diff
@@ -314,26 +335,20 @@ func run(cmd *cobra.Command, _ []string) {
 		os.Exit(1)
 	}
 
-	// Grab all the IDP information interactively if necessary
-	idpType := args.idpType
-	if idpType == "" {
-		interactive.Enable()
-	}
-
 	if interactive.Enabled() {
 		reporter.Infof("Interactive mode enabled.\n" +
 			"Any optional fields can be left empty and a default will be selected.")
 	}
 
-	if interactive.Enabled() {
-		if idpType == "" {
-			idpType = validIdps[0]
-		}
+	// Grab all the IDP information interactively if necessary
+	idpType := args.idpType
+	if idpType == "" {
+		interactive.Enable()
 		idpType, err = interactive.GetOption(interactive.Input{
 			Question: "Type of identity provider",
 			Options:  validIdps,
 			Required: true,
-			Default:  idpType,
+			Default:  validIdps[0],
```

Comment:
This is wrong. If a user runs `rosa create idp --type htpasswd --interactive` their prompts will revert back to `github`